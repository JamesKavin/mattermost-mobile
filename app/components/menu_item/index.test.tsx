// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {Preferences} from '@constants';
import {renderWithIntl} from '@test/intl-test-helper';

import MenuItem from '.';

describe('DrawerItem', () => {
    const baseProps = {
        onPress: () => null,
        testID: 'test-id',
        centered: true,
        defaultMessage: 'default message',
        i18nId: 'i18-id',
        iconName: 'icon-name',
        isDestructor: true,
        separator: true,
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = renderWithIntl(<MenuItem {...baseProps}/>);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should match snapshot without separator and centered false', () => {
        const props = {
            ...baseProps,
            centered: false,
            separator: false,
        };
        const wrapper = renderWithIntl(
            <MenuItem {...props}/>,
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
