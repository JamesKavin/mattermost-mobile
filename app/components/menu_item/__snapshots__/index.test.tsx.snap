// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerItem should match snapshot 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  testID="test-id"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "minHeight": 50,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 50,
              "justifyContent": "center",
              "marginLeft": 5,
              "width": 45,
            },
            undefined,
          ]
        }
      >
        <Icon
          name="icon-name"
          style={
            Array [
              Object {
                "color": "rgba(63,67,80,0.64)",
                "fontSize": 24,
              },
              Object {
                "color": "#d24b4e",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingBottom": 14,
              "paddingTop": 14,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(63,67,80,0.5)",
                  "fontSize": 17,
                  "includeFontPadding": false,
                  "textAlignVertical": "center",
                },
                Object {
                  "color": "#d24b4e",
                },
                Object {
                  "textAlign": "center",
                  "textAlignVertical": "center",
                },
              ]
            }
          >
            default message
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(63,67,80,0.2)",
              "height": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`DrawerItem should match snapshot without separator and centered false 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  testID="test-id"
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "minHeight": 50,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "height": 50,
              "justifyContent": "center",
              "marginLeft": 5,
              "width": 45,
            },
            undefined,
          ]
        }
      >
        <Icon
          name="icon-name"
          style={
            Array [
              Object {
                "color": "rgba(63,67,80,0.64)",
                "fontSize": 24,
              },
              Object {
                "color": "#d24b4e",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "justifyContent": "center",
              "paddingBottom": 14,
              "paddingTop": 14,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgba(63,67,80,0.5)",
                  "fontSize": 17,
                  "includeFontPadding": false,
                  "textAlignVertical": "center",
                },
                Object {
                  "color": "#d24b4e",
                },
                Object {},
              ]
            }
          >
            default message
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
